File: app/src/main/java/com/jsqix/dongqing/gank/adapter/recycler/YoukuProgramAdapter.java
package com.jsqix.dongqing.gank.adapter.recycler;

import android.content.Context;
import android.widget.ImageView;

import com.bumptech.glide.Glide;
import com.jsqix.dongqing.gank.R;
import com.jsqix.dongqing.gank.bean.youku.YKProgram;
import com.jsqix.dongqing.gank.utils.DateUtil;
import com.jsqix.dongqing.gank.utils.StringUtils;
import com.zhy.adapter.recyclerview.CommonAdapter;
import com.zhy.adapter.recyclerview.base.ViewHolder;

import java.util.List;

/**
 * Created by dongqing on 2017/1/16.
 */

public class YoukuProgramAdapter extends CommonAdapter<YKProgram.ShowsBean> {
    public YoukuProgramAdapter(Context context, int layoutId, List<YKProgram.ShowsBean> datas) {
        super(context, layoutId, datas);
    }

    @Override
    protected void convert(ViewHolder holder, YKProgram.ShowsBean showsBean, int position) {
        ImageView image = holder.getView(R.id.yk_image);
        Glide.with(mContext).load(showsBean.getPoster()).crossFade().into(image);
        holder.setText(R.id.tv_name, showsBean.getName());
        if ("电影".equals(showsBean.getCategory())) {
            holder.setVisible(R.id.layout_length, false);
        } else {
            holder.setVisible(R.id.layout_length, true);
        }
        if (StringUtils.toInt(showsBean.getCompleted()) == 1) {
            holder.setText(R.id.tv_length, showsBean.getEpisode_count() + "集全");
        } else {
            if (StringUtils.toInt(showsBean.getEpisode_count()) == 0) {
                if ("综艺".equals(showsBean.getCategory())) {
                    holder.setText(R.id.tv_length, "更新至" + (DateUtil.isDate(showsBean.getEpisode_updated(), "yyyyMMdd") ? showsBean.getEpisode_updated() : showsBean.getEpisode_updated() + "期"));
                } else {
                    holder.setText(R.id.tv_length, "更新至" + showsBean.getEpisode_updated() + "集");
                }
            } else {
                if (StringUtils.toInt(showsBean.getEpisode_updated()) == StringUtils.toInt(showsBean.getEpisode_count())) {
                    holder.setText(R.id.tv_length, showsBean.getEpisode_count() + "集全");
                } else {
                    holder.setText(R.id.tv_length, "更新至" + showsBean.getEpisode_updated() + "集");
                }
            }
        }
        if ("综艺".equals(showsBean.getCategory()) || "体育".equals(showsBean.getCategory()) || "纪录片".equals(showsBean.getCategory())) {
            holder.setText(R.id.tv_comment, "播放：" + StringUtils.toNumber(showsBean.getView_count()));
        } else {
            holder.setText(R.id.tv_comment, "评分：" + showsBean.getScore());
        }

    }
}

File: app/src/main/java/com/dongqing/gank/kotlin/adapter/YoukuProgramAdapter.kt
package com.dongqing.gank.kotlin.adapter

import android.content.Context
import android.content.Intent
import android.widget.ImageView
import com.dongqing.gank.kotlin.GlideApp
import com.dongqing.gank.kotlin.R
import com.dongqing.gank.kotlin.bean.youku.YKProgram
import com.dongqing.gank.kotlin.ui.activity.YoukuPlayerActivity
import com.dongqing.gank.kotlin.utils.DateUtil
import com.dongqing.gank.kotlin.utils.StringUtils
import com.zhy.adapter.recyclerview.CommonAdapter
import com.zhy.adapter.recyclerview.base.ViewHolder

class YoukuProgramAdapter : CommonAdapter<YKProgram.ShowsBean> {
    constructor(context: Context?, datas: List<YKProgram.ShowsBean>) : super(context, R.layout.item_recycler_yk_program, datas)

    override fun convert(holder: ViewHolder?, t: YKProgram.ShowsBean?, position: Int) {
        var image = holder!!.getView<ImageView>(R.id.yk_image)
        GlideApp.with(mContext).load(t!!.poster).into(image)
        holder.setText(R.id.tv_name, t.name)
        if ("电影" == t.category) {
            holder.setVisible(R.id.layout_length, false)
        } else {
            holder.setVisible(R.id.layout_length, true)
        }
        if (StringUtils.toInt(t.completed) === 1) {
            holder.setText(R.id.tv_length, t.episode_count + "集全")
        } else {
            if (StringUtils.toInt(t.episode_count) === 0) {
                if ("综艺" == t.category) {
                    holder.setText(R.id.tv_length, "更新至" + if(DateUtil.isDate(t.episode_updated, "yyyyMMdd")) t.episode_updated else t.episode_updated + "期")
                } else {
                    holder.setText(R.id.tv_length, "更新至" + t.episode_updated + "集")
                }
            } else {
                if (StringUtils.toInt(t.episode_updated) === StringUtils.toInt(t.episode_count)) {
                    holder.setText(R.id.tv_length, t.episode_count + "集全")
                } else {
                    holder.setText(R.id.tv_length, "更新至" + t.episode_updated + "集")
                }
            }
        }
        if ("综艺" == t.category || "体育" == t.category || "纪录片" == t.category) {
            holder.setText(R.id.tv_comment, "播放：" + StringUtils.toNumber(t.view_count))
        } else {
            holder.setText(R.id.tv_comment, "评分：" + t.score)
        }
        holder.itemView.setOnClickListener{
            val intent = Intent(mContext, YoukuPlayerActivity::class.java)
            intent.putExtra("id", t.id)
            intent.putExtra("url", t.play_link)
            mContext.startActivity(intent)
        }
    }
}